broker/bam/src/availability_builder.cc

```diff
@@ -59,13 +57,16 @@ availability_builder::~availability_builder() {}
  *  @param[in] end              The end of the event.
  *  @param[in] was_in_downtime  Was the event in downtime?
  *  @param[in] tp               The timeperiod of the event.
+ *  @param[in] logger           The logger to use.
  */
-void availability_builder::add_event(short status,
-                                     time_t start,
-                                     time_t end,
-                                     bool was_in_downtime,
-                                     time::timeperiod::ptr const& tp) {
-  log_v2::bam()->trace(
+void availability_builder::add_event(
+    short status,
+    time_t start,
+    time_t end,
+    bool was_in_downtime,
+    time::timeperiod::ptr const& tp,
+    const std::shared_ptr<spdlog::logger>& logger) {
```

Comment:
why not use an attribute as for other objects

Reply:
Because only this function uses the logger and only when trace is available... It will always be possible to embed it if needed.